### PR TITLE
Build with C++11 if gtkmm >= 3.18.0

### PR DESCRIPTION
**Pull request by jetm**
After Gnome and gtkmm 3.18 releases, build with C++11 is required.

### CODE REVIEW

**Comment by jetm** [comment]
It will help to solve issues like this: https://bugs.archlinux.org/task/46621

**Comment by andrewcchen** [comment]
Can you require c++11 on systems with gtkmm>=3.18. Sorry, I don't know how to do that as I am not familiar with autotools.

**Comment by jetm** [comment]
@andrewcchen: rewritten the commit to require C++11 when exists gtkmm >= 3.18.0. Thanks!

File: configure.ac [hunk 1]
```diff
@@ -413,6 +412,18 @@ then
...
     CPPFLAGS=$CPPFLAGS_save
+
+    with_cxx11=no
+    # Enable C++11 std if gtkmm >= 3.18.0
+    PKG_CHECK_MODULES(GTKMM, gtkmm-3.0 >= 3.18.0,
+      [ AX_CXX_COMPILE_STDCXX_11( [noext], [mandatory])
+        with_cxx11=yes
+      ],
+      [ AC_MSG_ERROR(
+	  [You need gtkmm version equal or newer than 3.18.0]
```
Comment (rcaelers) [inline configure.ac]: Workrave should still support gtkmm < 3.18. So I think this error message should be removed.

Comment (jetm) [inline configure.ac]: Removed the error message. Thanks @rcaelers

**Comment by rcaelers** [comment]
Thanks!

**Comment by pfrenssen** [comment]
:+1:

**Comment by zioalex** [comment]
On my gentoo system it didn't work out also with the latest workrave version 1.10.8.  I had to force the CXXFLAGS in the package.env files:
more package.env/workrave 
app-misc/workrave enablec++11.conf

more env/enablec++11.conf 
CXXFLAGS='${CFLAGS} --std=c++11'

Maybe if failed to determinate which gtkmm version I have because I have 2:
2.24.4-r3 and 3.18.0-r1

**Pull request merged**